[FEAT] Finalise Animal Interactions

spencerdezartsmith opened the pull request:
# Description

- Adds quick select to animal feeding
- Splits feeding into two actions, feed + collect
- Adds UI animation and sfx
- Add animal state images eg. sleeping, ready
- Actual numbers for food xp, drop amounts

![cow](https://github.com/user-attachments/assets/f4009209-dc21-4763-9e05-624faed0532b)
![chicken](https://github.com/user-attachments/assets/8cda1ab2-a65b-4fa4-8cc3-d379e0b74e25)


**EXCLUDES**
- Animal animations. Currently static images.
- Animal deals and bounties for barn animals (I have added the code but logic will need to be tweaked as both sheep and cows are inside the barn.

Fixes #issue

# What needs to be tested by the reviewer?

- Quick select animal foods
- Feed animals
- Level up and collect dropped resources

Please describe how this can be tested.

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

spencerdezartsmith commented on src/features/game/events/landExpansion/loveAnimal.ts:
```diff
@@ -81,6 +86,66 @@ const ANIMAL_ITEM_CHANCES: Record<
     Brush: 3,
     "Music Box": 0,
   },
+  4: {
```
These are dummy numbers. This is seperate to this work.

adamhannigan reviewed (approved):
Nice work mate! Loved the sound effects and in-game text. Just a small suggestion -Is it possible to animate the progress bar when it is increasing?

Also found one small bug:

<img width="118" alt="Screenshot 2024-10-21 at 10 23 29 AM" src="https://github.com/user-attachments/assets/6ad5a81d-95f6-4e1f-a11c-2d968443e6af">

spencerdezartsmith commented:
> Nice work mate! Loved the sound effects and in-game text. Just a small suggestion -Is it possible to animate the progress bar when it is increasing?
> 
> Also found one small bug:
> 
> <img alt="Screenshot 2024-10-21 at 10 23 29 AM" width="118" src="https://private-user-images.githubusercontent.com/11745561/378233021-6ad5a81d-95f6-4e1f-a11c-2d968443e6af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0NzE4MzAsIm5iZiI6MTcyOTQ3MTUzMCwicGF0aCI6Ii8xMTc0NTU2MS8zNzgyMzMwMjEtNmFkNWE4MWQtOTVmNi00ZTFmLWExMWMtMmQ5Njg0NDNlNmFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDAwNDUzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjZjlhY2RlMDRhZWI1ZWVkZjI2ZDNhNzIyMDZlZTg2ZDM1OThmNDFkYzMwOTM2NTU1YjMxOWQxM2UyODM1ZTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OCkenO5-O_4ZEQQKZdVJdfZzrw_bXbWnAkdXHhZ3j8I">

Have added an AnimatedBar. Also fixed the terms up. Also added and InfoPopover showing how long until the animal wakes up.
